pkg/apis/application/v1alpha1/types.go

```diff
@@ -55,7 +55,7 @@ type Application struct {
 // ApplicationSpec represents desired application state. Contains link to repository with application definition and additional parameters link definition revision.
 type ApplicationSpec struct {
 	// Source is a reference to the location of the application's manifests or chart
-	Source ApplicationSource `json:"source" protobuf:"bytes,1,opt,name=source"`
+	Source ApplicationSource `json:"source,omitempty" protobuf:"bytes,1,opt,name=source"`
```

Comment:
I've attempted to create an app with `sources` field only but noticed that controller injected empty `source` field after first reconciliation:

```
 project: default
  source:
    repoURL: ""
  sources:
  - path: guestbook
    repoURL: https://github.com/argoproj/argocd-example-apps
    targetRevision: HEAD
  - path: kustomize-guestbook
    repoURL: https://github.com/argoproj/argocd-example-apps
    targetRevision: HEAD 
```

I believe to fix it the `Source` field has to be a pointer.

Reply:
I've noticed that a lot of CLI code keep using `Source` field. I suggest introducing a method `func (s *ApplicationSpec) GetSource() ApplicationSource` that returns either Source field of first element of Sources. 

This way CLI methods would keep working even if `Source` field is missing and later we can improve them to visualize details of multiple sources.